---
title: "brain"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(cache = TRUE)
```
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r message=FALSE}
library(oligo)
library(pheatmap)
library(GEOquery)
require(gplots)
library(limma)
library(ggplot2)
library(readr)
library(dplyr)
library(sva)
library(ArrayExpress)
library(dplyr)
library(AnnotationDbi)
```

```{r}
#affymatrix from brain
#getGEOSuppFiles("GSE6955")
#untar("GSE6955/GSEGSE6955_RAW.tar", exdir = "GSE6955/CEL")
celfilesGSE6955 <- list.files("GSE6955/GSE6955_raw/CEL", full = TRUE)
GSE6955pheno <- read_csv("GSE6955pheno.csv")
SDRF <- AnnotatedDataFrame(GSE6955pheno)
GSE6955 <- read.celfiles(celfilesGSE6955,phenoData = SDRF)
```

```{r}
#checkqualityand batch
#pca
exp_raw <- log2(Biobase::exprs(GSE6955))
PCA_raw <- prcomp(t(exp_raw), scale. = FALSE)

percentVar <- round(100*PCA_raw$sdev^2/sum(PCA_raw$sdev^2),1)
sd_ratio <- sqrt(percentVar[2] / percentVar[1])

dataGG <- data.frame(PC1 = PCA_raw$x[,1], PC2 = PCA_raw$x[,2],
                    Disease = GSE6955@phenoData@data[["state"]]
                    )

ggplot(dataGG, aes(PC1, PC2)) +
      geom_point(aes(shape = Disease, colour = Disease)) +
  ggtitle("PCA plot of the log-transformed raw expression data") +
  xlab(paste0("PC1, VarExp: ", percentVar[1], "%")) +
  ylab(paste0("PC2, VarExp: ", percentVar[2], "%")) +
  theme(plot.title = element_text(hjust = 0.5))+
  coord_fixed(ratio = sd_ratio) 
```

```{r}
oligo::boxplot(GSE6955, target = "core", 
               main = "Boxplot of log2-intensitites for the raw data")
```
```{r}
distsRL <- dist(t(exp_raw))
mat <- as.matrix(distsRL)
rownames(mat) <- pData(GSE6955)$state
colnames(mat)<-pData(GSE6955)$state
hc <- hclust(distsRL)
heatmap.2(mat, Rowv=as.dendrogram(hc), symm=TRUE, trace="none", cexRow=1.0, cexCol=1.0, margin=c(13, 13), key=FALSE)
```

```{r}
#normalization and PCA
GSE6955_norm <- oligo::rma(GSE6955)

exp_palmieriID1 <- Biobase::exprs(GSE6955_norm)
PCA <- prcomp(t(exp_palmieriID1), scale = FALSE)

percentVar <- round(100*PCA$sdev^2/sum(PCA$sdev^2),1)
sd_ratio <- sqrt(percentVar[2] / percentVar[1])

dataGG <- data.frame(PC1 = PCA$x[,1], PC2 = PCA$x[,2],
                    Disease = 
                     Biobase::pData(GSE6955_norm)$state)


ggplot(dataGG, aes(PC1, PC2)) +
      geom_point(aes(shape = Disease, colour = Disease)) +
  ggtitle("PCA plot of the calibrated, summarized data") +
  xlab(paste0("PC1, VarExp: ", percentVar[1], "%")) +
  ylab(paste0("PC2, VarExp: ", percentVar[2], "%")) +
  theme(plot.title = element_text(hjust = 0.5)) +
  coord_fixed(ratio = sd_ratio) +
  scale_shape_manual(values = c(4,15)) + 
  scale_color_manual(values = c("darkorange2", "dodgerblue4"))

```

```{r}
#indicated good normalization
oligo::boxplot(GSE6955_norm, target = "core", 
               main = "Boxplot of log2-intensitites for the raw data")
```
```{r}
distsRL <- dist(t(exprs(GSE6955_norm)))
mat <- as.matrix(distsRL)
rownames(mat) <- pData(GSE6955_norm)$state
colnames(mat)<-pData(GSE6955_norm)$state
hc <- hclust(distsRL)
heatmap.2(mat, Rowv=as.dendrogram(hc), symm=TRUE, trace="none", cexRow=1.0, cexCol=1.0, margin=c(13, 13), key=FALSE)

```

```{r}
#filtering low intensity genes
GSE6955_f <- rowMedians(Biobase::exprs(GSE6955_norm))
```

```{r}
dev.off()
prop_cental <- 0.50
hist_res <- hist(GSE6955_f, 100, col = "cornsilk1", freq = FALSE, 
            main = "Histogram of the median intensities", 
            border = "antiquewhite4",
            xlab = "Median intensities")
				 
emp_mu <- hist_res$breaks[which.max(hist_res$density)]
emp_sd <- BiocGenerics::mad(GSE6955_f)/2
prop_cental <- 0.50
hist(GSE6955_f, 100, col = "cornsilk1", freq = FALSE, 
            main = "Histogram of the median intensities", 
            border = "antiquewhite4",
            xlab = "Medianintensities")


```

```{r}
#filtering 
cut_val <- 0.05 / prop_cental
thresh_median <- qnorm(0.05 / prop_cental, emp_mu, emp_sd)
no_of_samples <-table(paste0(GSE6955@phenoData@data[["state"]]))
no_of_samples
samples_cutoff <- min(no_of_samples)

idx_thresh_median <- apply(exprs(GSE6955_norm), 1, function(x){
				   sum(x > thresh_median) >= samples_cutoff})
table(idx_thresh_median)

```

```{r}
GSE6955_filtered <- subset(GSE6955_norm, idx_thresh_median)
#annotation to remove duplicates
library(hgu95av2.db)
GSE6955_anno <- AnnotationDbi::select(hgu95av2.db,
                                  keys =(featureNames(GSE6955_filtered)),
                                  columns = c("SYMBOL", "GENENAME"),
                                  keytype = "PROBEID")
probe_stats <- GSE6955_anno %>%
    group_by(PROBEID) %>%
    summarize(no_of_matches = n_distinct(SYMBOL)) %>%
    filter(no_of_matches > 1)
ids_to_exlude <- ((featureNames(GSE6955_filtered) %in% probe_stats$PROBEID) |
		 featureNames(GSE6955_filtered)  %in% subset(GSE6955_anno ,
							         is.na(SYMBOL))$PROBEID)
table(ids_to_exlude)

GSE6955_final <- subset(GSE6955_filtered, !ids_to_exlude)

fData(GSE6955_final)$PROBEID <- rownames(fData(GSE6955_final))
fData(GSE6955_final) <- left_join(fData(GSE6955_final), GSE6955_anno)

rownames(fData(GSE6955_final)) <-fData(GSE6955_final)$PROBEID

validObject(GSE6955_final)
```

```{r}
library(MetaIntegrator)
#creating metaintegrator list
caseGSE6955 <- read_csv("GSE6955/caseGSE6955.csv")
ID1<-list()
ID1$formattedName<-'RTT'
GSE6955batch<-GSE6955@protocolData@data[["dates"]]
GSE6955batch<-caseGSE6955$batch
GSE6955_filteredb<-ComBat(dat=exprs(GSE6955_final),batch=GSE6955batch,mod = caseGSE6955$control.0.col)
ID1$class<-setNames(caseGSE6955$state,colnames(GSE6955_filteredb))

ID1$expr<-GSE6955_filteredb
ID1$pheno<-caseGSE6955
g103965<-GSE6955_final@featureData@data[["SYMBOL"]]
keys103<-setNames(g103965, GSE6955_final@featureData@data[["PROBEID"]])
ID1$keys<-keys103
checkDataObject(ID1, "Dataset")
```

```{r}
#ID2 illumina
#IDGSE75303 Illumina HumanHT-12 V4.0 expression beadchip

#getGEOSuppFiles("GSE75303")
#there was this error "Error in gregexpr("\t", dataLine1)[[1]] : subscript out of bounds" when I used lumi so I searched and it was due to different structure of data on GEOdataset so I used non-normalized raw data
##############################
#read in the data and convert the data to an EListRaw object, which is a data object for single channel data
x <- lapply(
  list.files('GSE75303', pattern = '\\.txt.gz$',  full.names = TRUE),
  function(x) data.table::fread(file = x))
x <- do.call(cbind, x)
#annotation
annot <- read_csv("annot.csv")
annot <- annot[match(x$ID_REF, annot$ID),]
detectionpvalues<-x[,c(4,6,8,10,12,14,16,18,20,22,24,26)]
probes <- x$ProbeID
x[,c(4,6,8,10,12,14,16,18,20,22,24,26)]<-NULL
x <- data.matrix(x[,3:ncol(x)])
rownames(x) <- probes
colnames(x) <- c("GSM1949097","GSM1949098","GSM1949099","GSM1949100","GSM1949101",	"GSM1949102",	"GSM1949103",	"GSM1949104",	"GSM1949105",	"GSM1949106",	"GSM1949107",	"GSM1949108")

# create a custom EListRaw object
targetsfile <- read_csv("targetsfile.csv")
project <- new('EListRaw')
project$E <- x
project$targets <- targetsfile
project$genes <- annot
 project$other$Detection <- detectionpvalues
#checkingdata
#boxplot
boxplot(project$E)
```

```{r}
#PCA
exp_raw <- log2(project$E)
PCA_raw <- prcomp(t(exp_raw), scale. = FALSE)

percentVar <- round(100*PCA_raw$sdev^2/sum(PCA_raw$sdev^2),1)
sd_ratio <- sqrt(percentVar[2] / percentVar[1])

dataGG <- data.frame(PC1 = PCA_raw$x[,1], PC2 = PCA_raw$x[,2],
                     Disease = targetsfile$status)

ggplot(dataGG, aes(PC1, PC2)) +
  geom_point(aes(shape = Disease, colour = Disease)) +
  ggtitle("PCA plot of the log-transformed raw expression data") +
  xlab(paste0("PC1, VarExp: ", percentVar[1], "%")) +
  ylab(paste0("PC2, VarExp: ", percentVar[2], "%")) +
  theme(plot.title = element_text(hjust = 0.5))+
  coord_fixed(ratio = sd_ratio) 
```
```{r}
project.bgcorrect.norm <- neqc(project)
boxplot(project.bgcorrect.norm$E)
```
```{r}
#PCA
exp_raw <- project.bgcorrect.norm$E
PCA_raw <- prcomp(t(exp_raw), scale. = FALSE)

percentVar <- round(100*PCA_raw$sdev^2/sum(PCA_raw$sdev^2),1)
sd_ratio <- sqrt(percentVar[2] / percentVar[1])

dataGG <- data.frame(PC1 = PCA_raw$x[,1], PC2 = PCA_raw$x[,2],
                     Disease = targetsfile$status)

ggplot(dataGG, aes(PC1, PC2)) +
  geom_point(aes(shape = Disease, colour = Disease)) +
  ggtitle("PCA plot of the log-transformed raw expression data") +
  xlab(paste0("PC1, VarExp: ", percentVar[1], "%")) +
  ylab(paste0("PC2, VarExp: ", percentVar[2], "%")) +
  theme(plot.title = element_text(hjust = 0.5))+
  coord_fixed(ratio = sd_ratio) 


```

```{r}
# filter out control probes, those with no symbol, and those with low expression
Control <- project.bgcorrect.norm$genes$Source=="ILMN_Controls"
NoSymbol <- project.bgcorrect.norm$genes$Symbol == ""
isexpr <- rowSums(project.bgcorrect.norm$other$Detection <= 0.05) >= 3 #3 as in limma package
project.bgcorrect.norm.filt <- project.bgcorrect.norm[!Control & !NoSymbol & isexpr, ]
dim(project.bgcorrect.norm)
dim(project.bgcorrect.norm.filt)

# remove annotation columns we no longer need
project.bgcorrect.norm.filt$genes <- project.bgcorrect.norm.filt$genes[,c(
    'ID',
    'Definition','Ontology_Component','Ontology_Process','Ontology_Function',
    'Chromosome','Probe_Coordinates','Cytoband','Probe_Chr_Orientation',
    'RefSeq_ID','Entrez_Gene_ID','Symbol')]
head(project.bgcorrect.norm.filt$genes)

# summarise across genes by mean
# ID is used to identify the replicates
project.bgcorrect.norm.filt.mean <- avereps(project.bgcorrect.norm.filt$E,
    ID = project.bgcorrect.norm.filt$genes$Symbol)
  
dim(project.bgcorrect.norm.filt.mean)
#######################
caseGSE75303 <- read_csv("targetsfile.csv")
a<-as.data.frame.matrix(project.bgcorrect.norm.filt.mean)
v<-data.matrix(a, rownames.force = NA)
row.names(caseGSE75303)<-caseGSE75303$Samples
colnames(v)<-row.names(caseGSE75303)
ID2<-list()
ID2$formattedName<-'RETT'
ID2$class<-setNames(caseGSE75303$value,colnames(v))
b<-colnames(v)
c<-caseGSE75303$value
d<-cbind(b,c)
ID2$expr<-v
ID2$pheno<-caseGSE75303
#g85804<-row.names(project.bgcorrect.norm.filt.mean)
#annotf <- annot[match(annot$Symbol,row.names(project.bgcorrect.norm.filt.mean)),]
keys8580<-setNames(row.names(a),row.names(a))#just for the package as it is already annotated
ID2$keys<-keys8580
checkDataObject(ID2, "Dataset")
#####

```
```{r}
#ID3
raw_data_dir <- tempdir()

if (!dir.exists(raw_data_dir)) {
    dir.create(raw_data_dir)
}
anno_AE <- getAE("E-MEXP-3288", path = raw_data_dir, type = "raw")
sdrf_location <- file.path(raw_data_dir, "E-MEXP-3288.sdrf.txt")
SDRF <- read.delim(sdrf_location)
rownames(SDRF) <- SDRF$Array.Data.File
SDRF <- AnnotatedDataFrame(SDRF)
raw_data <- oligo::read.celfiles(filenames = file.path(raw_data_dir,
                                                      SDRF$Array.Data.File),
                                         verbose = FALSE, phenoData = SDRF)
validObject(raw_data)
```
```{r}
head(Biobase::pData(raw_data))

```

```{r}

exp_raw <- log2(Biobase::exprs(raw_data))
PCA_raw <- prcomp(t(exp_raw), scale. = FALSE)
percentVar <- round(100*PCA_raw$sdev^2/sum(PCA_raw$sdev^2),1)
sd_ratio <- sqrt(percentVar[2] / percentVar[1])
dataGG <- data.frame(PC1 = PCA_raw$x[,1], PC2 = PCA_raw$x[,2],
                       Disease = raw_data@phenoData@data[["Characteristics.Genotype."]])
ggplot(dataGG, aes(PC1, PC2)) +
       geom_point(aes(shape = Disease,colour=Disease)) +
  ggtitle("PCA plot of the log-transformed raw expression data") +
  xlab(paste0("PC1, VarExp: ", percentVar[1], "%")) +
  ylab(paste0("PC2, VarExp: ", percentVar[2], "%")) +
  theme(plot.title = element_text(hjust = 0.5))+
  coord_fixed(ratio = sd_ratio)
```

```{r}
oligo::boxplot(raw_data, target = "core",
                 main = "Boxplot of log2-intensitites for the raw data")
```

```{r}
palmieri_eset_norm <- oligo::rma(raw_data)
```
```{r}
exp_palmieri <- Biobase::exprs(palmieri_eset_norm)
PCA <- prcomp(t(exp_palmieri), scale = FALSE)

percentVar <- round(100*PCA$sdev^2/sum(PCA$sdev^2),1)
sd_ratio <- sqrt(percentVar[2] / percentVar[1])

dataGG <- data.frame(PC1 = PCA$x[,1], PC2 = PCA$x[,2],
                      Disease =raw_data@phenoData@data[["Characteristics.Genotype."]])


ggplot(dataGG, aes(PC1, PC2)) +
       geom_point(aes(shape = Disease, colour = Disease)) +
  ggtitle("PCA plot of the calibrated, summarized data") +
  xlab(paste0("PC1, VarExp: ", percentVar[1], "%")) +
  ylab(paste0("PC2, VarExp: ", percentVar[2], "%")) +
  theme(plot.title = element_text(hjust = 0.5)) +
  coord_fixed(ratio = sd_ratio) +
  scale_shape_manual(values = c(4,15)) +
  scale_color_manual(values = c("darkorange2", "dodgerblue4"))
```

```{r}

dists <- as.matrix(dist(t(exp_palmieri), method = "manhattan"))

rownames(dists) <- row.names(pData(palmieri_eset_norm))
hmcol <- rev(colorRampPalette(RColorBrewer::brewer.pal(9, "YlOrRd"))(255))
colnames(dists) <- NULL
diag(dists) <- NA

ann_colors <- list(Disease = c(control = "blue4", XALD = "cadetblue2")
            )

pheatmap(dists, col = (hmcol),
          
          annotation_colors = ann_colors,
          legend = TRUE,
          treeheight_row = 0,
          legend_breaks = c(min(dists, na.rm = TRUE),
                            max(dists, na.rm = TRUE)),
          legend_labels = (c("small distance", "large distance")),
          main = "Clustering heatmap for the calibrated samples")
```

```{r}
#filtering low intensity genes
palmieri_medians <- rowMedians(Biobase::exprs(palmieri_eset_norm))
```

```{r}
dev.off()
prop_cental <- 0.50
hist_res <- hist(palmieri_medians, 100, col = "cornsilk1", freq = FALSE, 
            main = "Histogram of the median intensities", 
            border = "antiquewhite4",
            xlab = "Median intensities")
				 
emp_mu <- hist_res$breaks[which.max(hist_res$density)]
emp_sd <- BiocGenerics::mad(palmieri_medians)/2
prop_cental <- 0.50
hist(palmieri_medians, 100, col = "cornsilk1", freq = FALSE, 
            main = "Histogram of the median intensities", 
            border = "antiquewhite4",
            xlab = "Medianintensities")


```

```{r}

#filtering 
cut_val <- 0.05 / prop_cental
thresh_median <- qnorm(0.05 / prop_cental, emp_mu, emp_sd)
no_of_samples <-table(paste0(raw_data@phenoData@data[["Characteristics.Genotype."]]))
no_of_samples
samples_cutoff <- min(no_of_samples)

idx_thresh_median <- apply(exprs(palmieri_eset_norm), 1, function(x)
  {sum(x > thresh_median) >= samples_cutoff})

table(idx_thresh_median)
palmieri_filtered <- subset(palmieri_eset_norm, idx_thresh_median)
```

```{r}
#annotation to remove duplicates
library(hgu133plus2.db)

anno_palmieri <- AnnotationDbi::select(hgu133plus2.db,
                                       keys =(featureNames(palmieri_filtered)),
                                       columns = c("SYMBOL", "GENENAME"),
                                       keytype = "PROBEID")
probe_stats <- anno_palmieri %>%
    group_by(PROBEID) %>%
    summarize(no_of_matches = n_distinct(SYMBOL)) %>%
    filter(no_of_matches > 1)

ids_to_exlude <- ((featureNames(palmieri_filtered) %in% probe_stats$PROBEID) |
		 featureNames(palmieri_filtered)  %in% subset(anno_palmieri,
							         is.na(SYMBOL))$PROBEID)
table(ids_to_exlude)

MEXP_final <- subset(palmieri_filtered, !ids_to_exlude)

fData(MEXP_final)$PROBEID <- rownames(fData(MEXP_final))
fData(MEXP_final) <- left_join(fData(MEXP_final), anno_palmieri)
rownames(fData(MEXP_final)) <-fData(MEXP_final)$PROBEID
validObject(MEXP_final)
```
```{r}
################

library(MetaIntegrator)
MEXP <- read_csv("MEXP2.csv")
row.names(MEXP)<-MEXP$Samples
ID3<-list()
ID3$formattedName<-'Adrenoleukodystrophy'
statw<-c("0","0","0","0","0","0","0","0","0","0","0","1","1","1","1","1","1","1","1","1","1","1","1","1")
MEXP_final@protocolData@data[["dates"]]
MEXPbatch<-c("B1","B1","B1","B1","B1","B1","B1","B1","B1","B1","B1","B1","B1","B1","B1","B1","B1","B1","B1","B1","B1","B1","B2","B1")

MEXP_filteredb<-ComBat(dat=exprs(MEXP_final), batch=MEXPbatch)
ID3$class<-setNames(MEXP$C,colnames(MEXP_filteredb))

ID3$expr<-MEXP_filteredb
ID3$pheno<-MEXP
g34308<-MEXP_final@featureData@data[["SYMBOL"]]
keys210<-setNames(g34308, MEXP_final@featureData@data[["PROBEID"]])
ID3$keys<-keys210
checkDataObject(ID3, "Dataset")
```

```{r}
#finaland create the metaObject
discovery_datasets <- list(ID1, ID2,ID3)
names(discovery_datasets) = c(ID1$formattedName, ID2$formattedName,ID3$formattedName)
MetaObj=list() 
MetaObj$originalData <- discovery_datasets
checkDataObject(MetaObj, "Meta", "Pre-Analysis")
library(MetaIntegrator)
finalll<-geneSymbolCorrection(MetaObj)
sMetaObj <- runMetaAnalysis(finalll,maxCores = 1)
```

```{r}
#filter at FDR=0.05
leMetaAnalysis <- filterGenes(sMetaObj, isLeaveOneOut = F, effectSizeThresh = 1, FDRThresh = 0.05)
#extraction of effect size
effectsize<-data.table::data.table(summarizeFilterResults(leMetaAnalysis, "FDR0.05_es1_nStudies1_looaFALSE_hetero0"))
```

```{r}
#forwardsearch for more accurate selection (featureselection)
forwardRes <- forwardSearch(leMetaAnalysis, leMetaAnalysis$filterResults[[1]], forwardThresh = 0)
up<-effectsize[[1]][[1]]
down<-effectsize[[1]][[2]]
positive<-forwardRes[["posGeneNames"]]
negative<-forwardRes[["negGeneNames"]]
write.csv(positive,"forwardfilteredup.csv")
write.csv(negative,"forwardfiltereddown.csv")
#subsetting the up and down regulating gene after feature selection
upfiltered<-subset(up, rownames(up) %in% positive)
downfilterd<-subset(down, rownames(down) %in% negative)
write.csv(upfiltered,"upregulated.csv")
write.csv(downfilterd,"downregulated.csv")
#converted filtered genes to UniProtKB 
#open verified O-GlcNAC database, and extracting only genes coding for proteins undergoing O-GlcNAc
#verified O-glcnac proteins
Oglcnac <- read_csv("Oglcnac.csv")
#uniprotID of genes
upregulateduniprot <- read_csv("upregulateduniprot.csv")
#downuniprot <- mapIds(org.Hs.eg.db,keys = negative,column = "UNIPROT",keytype = "SYMBOL",multiVals = "first")
#determinning which genes are coding for O-GlcNAcylated proteins
upp<-subset(Oglcnac, Oglcnac$`UniprotKB ID` %in% upregulateduniprot$Entry)
#downn<-subset(Oglcnac, Oglcnac$`UniprotKB ID` %in% downuniprot)
write.csv(upp,"final_up-ogl.csv")
library(org.Hs.eg.db)
library(clusterProfiler)
#convert uniprot ito entrez ID

upp$entrez <- mapIds(org.Hs.eg.db,
                     keys = upp$`UniprotKB ID`,
                     column = "ENTREZID",
                     keytype = "UNIPROT",
                     multiVals = "first")
upp$gene<-mapIds(org.Hs.eg.db,
                     keys = upp$`UniprotKB ID`,
                     column = "SYMBOL",
                     keytype = "UNIPROT",
                     multiVals = "first")
#downn$gene<-mapIds(org.Hs.eg.db,keys = downn$`UniprotKB ID`,
#column = "SYMBOL",
#keytype = "UNIPROT",
#multiVals = "first")

#backgroundgenes, using all meta-analysis genes
hj<-mapIds(org.Hs.eg.db,
                     keys =positive,
                     column = "ENTREZID",
                     keytype = "SYMBOL",
                     multiVals = "first")
#using upregulated DEG genes as background genes
ego <- enrichGO(gene          = upp$entrez,
                universe = as.character(hj),
                OrgDb         = org.Hs.eg.db,
                ont           = "CC",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05,
                 readable      = TRUE)
dotplot(ego)
```
```{r}
ego <- enrichGO(gene          = upp$entrez,
                universe = as.character(hj),
                OrgDb         = org.Hs.eg.db,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05,
                 readable      = TRUE)
dotplot(ego)
#write.csv(downn,"final_down-ogl.csv")
head(upp)
```

```{r}
ego <- enrichGO(gene          = upp$entrez,
                universe = as.character(hj),
                OrgDb         = org.Hs.eg.db,
                ont           = "MF",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05,
                 readable      = TRUE)
dotplot(ego)
```
```{r}
library(DOSE)
de<-upp$entrez
edo <- enrichDO(de)
library(enrichplot)
barplot(edo, showCategory=20)
```

```{r}
#subsetting downregulted genes
downfilterd<-subset(down, rownames(down) %in% negative)
downfilterd$uniprot<- mapIds(org.Hs.eg.db,
                     keys = row.names(downfilterd),
                     column = "UNIPROT",
                     keytype = "SYMBOL",
                     multiVals = "first")
#subsetting genes coding for O-GlcNAcylated proteins to start GO enrichment
downn<-subset(Oglcnac, Oglcnac$`UniprotKB ID` %in% downfilterd$uniprot)
#GOenrich
#convert uniprot ito entrez ID

downn$entrez <- mapIds(org.Hs.eg.db,
                     keys = downn$`UniprotKB ID`,
                     column = "ENTREZID",
                     keytype = "UNIPROT",
                     multiVals = "first")

#using downregulated DEG genes as background genes
hjn<-mapIds(org.Hs.eg.db,
                     keys =negative,
                     column = "ENTREZID",
                     keytype = "SYMBOL",
                     multiVals = "first")
ego <- enrichGO(gene          = downn$entrez,
                universe = as.character(hjn),
                OrgDb         = org.Hs.eg.db,
                ont           = "CC",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05,
                 readable      = TRUE)
dotplot(ego)
```
```{r}
ego <- enrichGO(gene          = downn$entrez,
                universe = as.character(hjn),
                OrgDb         = org.Hs.eg.db,
                ont           = "MF",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05,
                 readable      = TRUE)
dotplot(ego)
```
```{r}
ego <- enrichGO(gene          = downn$entrez,
                universe = as.character(hjn),
                OrgDb         = org.Hs.eg.db,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05,
                 readable      = TRUE)
dotplot(ego)
```
```{r}
edo <- enrichDO(downn$entrez)
barplot(edo, showCategory=20)
```

```{r}
#Go enrichment for all UPREGULATED AND DOWNREGULATED DEG
upreg<-mapIds(org.Hs.eg.db,
                     keys =positive,
                     column = "ENTREZID",
                     keytype = "SYMBOL",
                     multiVals = "first")
downreg<-mapIds(org.Hs.eg.db,
                     keys =negative,
                     column = "ENTREZID",
                     keytype = "SYMBOL",
                     multiVals = "first")
Backgoundgenes<-row.names(leMetaAnalysis[["metaAnalysis"]][["datasetEffectSizes"]])
BG<-mapIds(org.Hs.eg.db,
                     keys =Backgoundgenes,
                     column = "ENTREZID",
                     keytype = "SYMBOL",
                     multiVals = "first")

```
```{r}
#GO enrichment up DEG
ego <- enrichGO(gene          = upreg,
                universe = BG,
                OrgDb         = org.Hs.eg.db,
                ont           = "cc",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05,
                 readable      = TRUE)
dotplot(ego)
```


```{r}
ego <- enrichGO(gene          = downreg,
                universe = BG,
                OrgDb         = org.Hs.eg.db,
                ont           = "cc",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05,
                 readable      = TRUE)
dotplot(ego)
```
```{r}
ego <- enrichGO(gene          = downreg,
                universe = BG,
                OrgDb         = org.Hs.eg.db,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05,
                 readable      = TRUE)
dotplot(ego)
```
```{r}
ego <- enrichGO(gene          = downreg,
                universe = BG,
                OrgDb         = org.Hs.eg.db,
                ont           = "MF",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05,
                 readable      = TRUE)
dotplot(ego)
```
```{r}
edo <- enrichDO(downreg)
barplot(edo, showCategory=20)
```
```{r}

h<-leMetaAnalysis
h[["filterResults"]][["FDR0.05_es1_nStudies1_looaFALSE_hetero0"]][["posGeneNames"]]<-upp$gene
h[["filterResults"]][["FDR0.05_es1_nStudies1_looaFALSE_hetero0"]][["negGeneNames"]]<-downn$gene

lincsHits <- lincsCorrelate( metaObject = h, filterObject = h$filterResults[[1]], dataset = "CP", direction = "reverse",cor.method = "pearson")
```
```{r}
lincsHits <- lincsCorrelate( metaObject = h, filterObject = h$filterResults[[1]], dataset = "SH", direction = "reverse",cor.method = "pearson")
```
```{r}
lincsHits <- lincsCorrelate( metaObject = sMetaObj, filterObject = leMetaAnalysis$filterResults[[1]], dataset = "LIG", direction = "reverse",cor.method = "pearson")
```

```{r}
#other annotation analysis
library(magrittr)
library(clusterProfiler)
library(DOSE)
library(AnnotationDbi)
library(org.Hs.eg.db)
g<-as.data.frame(modGenes)
g$y<-mapIds(org.Hs.eg.db,
                     keys = modGenes,
                     column = "ENTREZID",
                     keytype = "SYMBOL",
                     multiVals = "first")
hj<-as.data.frame(colnames(datExpr))
hj$y<-mapIds(org.Hs.eg.db,
                     keys = colnames(datExpr),
                     column = "ENTREZID",
                     keytype = "SYMBOL",
                     multiVals = "first")
#GO enrichment
ego <- enrichGO(gene          = g$y,OrgDb         = org.Hs.eg.db,
                keyType       = 'ENTREZID',
              ont           = "MF",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.05)
dotplot(x)
ego3 <- gseGO(geneList     = g$y,
              OrgDb        = org.Hs.eg.db,
              ont          = "CC",
              nPerm        = 1000,
              minGSSize    = 100,
              maxGSSize    = 500,
              pvalueCutoff = 0.05,
              verbose      = FALSE)
goplot(ego)
library(enrichplot)
library(clusterProfiler)
library(DOSE)
barplot(edo, showCategory=20)
edo <- enrichDGN(g$y)
edox <- setReadable(edo, 'org.Hs.eg.db', 'ENTREZID')
p1 <- cnetplot(edox, foldChange=geneList)
## categorySize can be scaled by 'pvalue' or 'geneNum'
p2 <- cnetplot(edox, categorySize="pvalue", foldChange=geneList)
p3 <- cnetplot(edox, foldChange=geneList, circular = TRUE, colorEdge = TRUE)
cowplot::plot_grid(p1, p2, p3, ncol=3, labels=LETTERS[1:3], rel_widths=c(.8, .8, 1.2))
p1 <- cnetplot(edox, node_label="category") 
p2 <- cnetplot(edox, node_label="gene") 
p3 <- cnetplot(edox, node_label="all") 
p4 <- cnetplot(edox, node_label="none") 
cowplot::plot_grid(p1, p2, p3, p4, ncol=2, labels=LETTERS[1:4])

p1 <- heatplot(edox)
upsetplot(edo)

#survivall analysis
#survdata<-getSampleLevelGeneData(leMetaAnalysis$originalData[[2]], #geneNames=c(leMetaAnalysis$filterResults[[1]]$posGeneNames, 
#leMetaAnalysis$filterResults[[1]]$negGeneNames))
#survdata$gene<-row.names(s
urvdata)
#h<-as.data.frame(project.bgcorrect.norm.filt.mean)
#h$genes<-row.names(h)
#h2 <- h[,match(h$genes,leMetaAnalysis$filterResults[[1]]$negGeneNames)]

```
